freeipa/src/main/java/com/sequenceiq/freeipa/service/filter/GcpNetworkFilterProvider.java

```diff
@@ -0,0 +1,45 @@
+package com.sequenceiq.freeipa.service.filter;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.stereotype.Service;
+
+import com.google.common.base.Strings;
+import com.sequenceiq.cloudbreak.common.mappable.CloudPlatform;
+import com.sequenceiq.freeipa.entity.Network;
+
+@Service
+public class GcpNetworkFilterProvider implements NetworkFilterProvider {
+
+    @Override
+    public Map<String, String> provide(Network network, String networkId, Collection<String> subnetIds) {
+        Map<String, String> filter = new HashMap<>();
+        if (network.getAttributes() != null && network.getAttributes().getMap() != null) {
+            Map<String, Object> attributes = network.getAttributes().getMap();
+            String sharedProjectId = (String) attributes.get("sharedProjectId");
+            if (!Strings.isNullOrEmpty(sharedProjectId)) {
+                filter.put("sharedProjectId", sharedProjectId);
+            }
+        }
+        filter.put("networkId", networkId);
+        buildSubnetIdFilter(subnetIds, filter);
+        return filter;
+    }
+
+    @Override
+    public CloudPlatform cloudPlatform() {
+        return CloudPlatform.GCP;
+    }
+
+    private void buildSubnetIdFilter(Collection<String> subnetIds, Map<String, String> filter) {
```

Comment:
it would be much nicer if it would return with like a map and we could use `filter.putAll` where it is called instead of modifying the map passed via a parameter. It could return with empty map if not applicable